lib/mozart_fetcher/timeout_parser.ex

```diff
@@ -0,0 +1,37 @@
+defmodule MozartFetcher.TimeoutParser do
+  @default_timeout MozartFetcher.content_timeout()
+
+  def parse(endpoint) do
+    uri = URI.parse(endpoint)
+    query(uri.query)
+  end
+
+  defp query(nil) do
+    @default_timeout
+  end
+
+  defp query(q) do
+    qs = URI.decode_query(q)
+    fetch(qs["timeout"])
+  end
+
+  defp fetch(nil) do
+    @default_timeout
+  end
+
+  defp fetch("0") do
+    @default_timeout
+  end
+
+  defp fetch(qs_timeout) do
```

Comment:
just checking this is always guaranteed to be a string? Because `Integer.parse(<integer>)` will throw an error

Reply:
it _should_ as `URI.parse` always produces qs values as string

Reply:
I have wrapped that earlier in `parse_timeout` which could handle an error. If so, this will just be the default timeout.

`URI.parse` could produce a nil for the qs value, but other than that it will be a string. If we use Integer.parse on a string such as "abc" then this will just be replaced with the default timeout.

There are tests for all of these situations.